x/claim/keeper/claim.go

```diff
@@ -0,0 +1,235 @@
+package keeper
+
+import (
+	"errors"
+	"strings"
+
+	"github.com/arkeonetwork/arkeo/x/claim/types"
+	"github.com/cosmos/cosmos-sdk/store/prefix"
+	sdk "github.com/cosmos/cosmos-sdk/types"
+	sdkerrors "github.com/cosmos/cosmos-sdk/types/errors"
+	"github.com/gogo/protobuf/proto"
+)
+
+// SetClaimRecord sets a claim record for an address in store
+func (k Keeper) SetClaimRecord(ctx sdk.Context, claimRecord types.ClaimRecord) error {
+	store := ctx.KVStore(k.storeKey)
+	prefixStore := prefix.NewStore(store, chainToStorePrefix(claimRecord.Chain))
+
+	bz, err := proto.Marshal(&claimRecord)
+	if err != nil {
+		return err
+	}
+
+	addr := []byte(strings.ToLower(claimRecord.Address))
+
+	// validate address if valid based on chain
+	switch claimRecord.Chain {
+	case types.ETHEREUM:
+		if !types.IsValidEthAddress(claimRecord.Address) {
+			return sdkerrors.Wrapf(sdkerrors.ErrInvalidAddress, "invalid eth address (%s)", err)
+		}
+	case types.ARKEO:
+		_, err := sdk.AccAddressFromBech32(claimRecord.Address)
+		if err != nil {
+			return sdkerrors.Wrapf(sdkerrors.ErrInvalidAddress, "invalid arkeo address (%s)", err)
+		}
+	case types.THORCHAIN:
+		return errors.New("thorchain not supported yet")
+	}
+
+	prefixStore.Set(addr, bz)
```

Comment:
I feel use `addr` as the key is not good , because in the future , if you support other chains.
for example, ETH , Avalanche, BSC , their address all looks the same...